shadowsocks-csharp/Model/Server.cs

```diff
@@ -43,6 +43,7 @@ public class Server
         [DefaultValue("")]
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public string plugin_args;
+        // TODO The legacy of the historical version (Required field in SIP003)

```

Comment:
Is `remarks` a required field?

Reply:
according to SIP002 spec: `SS-URI = "ss://" userinfo "@" hostname ":" port [ "/" ] [ "?" plugin ] [ "#" tag ]`

Reply:
From the description of `Basic file format`. Yes

Reply:
In fact, I want to initiate a PR to your fork, but I didn't expect that the Desktop program was pushed directly to the branch.

Reply:
https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/doc-json.md#profile-json-object

> A JSON object is recognized as a `Profile` (i.e. a valid server config), if all of the following required fields are present and valid:
>
> - `server`
> - `server_port`
> - `password`
> - `method`

Reply:
Oh, it looks like the two descriptions conflict.

https://github.com/shadowsocks/shadowsocks-org/issues/89

> ## Basic file format
> Basically, we follows the original JSON config format of shadowsocks. The required fields are:
> 
> ```json
> {
>   "server": "198.199.101.152",
>   "server_port": 8388,
>   "password": "u1rRWTssNv0p",
>   "method": "aes-256-cfb",
>   "remarks": "Example 1"
> }
> ```

Reply:
It's confused, we should discuss it in that issue.

Reply:
Let [doc-json.md](https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/doc-json.md#profile-json-object) prevail. For maximum compatibility

Reply:
You can delete `TODO` and update the basic format , then force push.

Reply:
sounds good.

> In fact, I want to initiate a PR to your fork, but I didn't expect that the Desktop program was pushed directly to the branch.

BTW, I personally recommend GitExtension.